sentinel.py

```diff
@@ -4,31 +4,50 @@
 """
 Create sentinel and singleton objects.
 
-Copyright 2014 © Eddie Antonio Santos. MIT licensed.
+ Copyright 2014 © Eddie Antonio Santos. MIT licensed.
+
+ This software is released under the MIT License.
+ https://opensource.org/licenses/MIT
 """
 
 import inspect
+from typing import Dict, Tuple, Optional, Any
 
 __all__ = ['create']
-__version__ = '0.1.2'
+__version__ = '0.1.3'
 
+class _SentinelMeta(type):
+    def __call__(cls, *args:Tuple[Any], **kwargs:Dict[str, Any]) -> object:
+        inst = _sinstances.get(cls)
+        if inst is None:
+            inst = type.__call__(cls, *args, **kwargs)
+            _sinstances[cls] = inst
 
-def get_caller_module():
-    """
-    Returns the name of the caller's module as a string.
+        return inst
 
-    >>> get_caller_module()
-    '__main__'
-    """
+_sinstances: Dict[_SentinelMeta, object] = {}
+_SentinelMeta.instance = property(fget=_sinstances.get, doc="""Gets the instance of this sentinel type""")
+
+__mro_immortal = (object, )
```

Comment:
I'd just like to point out that I googled if this is an existing naming convention Google said, "Did you mean _my immortal_?"

![Screen Shot 2020-12-30 at 9 29 16 AM](https://user-images.githubusercontent.com/2294397/103367157-9c647d00-4a81-11eb-89c0-bcacf134f218.png)


I also find it quaint that Google thinks My Immortal, the infamous fanfic is a "book".

---

Incidentally, I'd use a naming convention that PyLint would actually accept!

```suggestion
_DEFAULT_MRO = (object, )
```